ansible/roles/test/files/ptftests/inner_hash_test.py

```diff
@@ -0,0 +1,505 @@
+    def setUp(self):
+        '''
+        @summary: Setup for the test
+        '''
+        self.dataplane = ptf.dataplane_instance
+
+        self.fib = fib.Fib(self.test_params['fib_info'])
+        self.router_mac = self.test_params['router_mac']
+
+        inner_src_ip_range = [unicode(x) for x in self.test_params['inner_src_ip_range'].split(',')]
+        inner_dst_ip_range = [unicode(x) for x in self.test_params['inner_dst_ip_range'].split(',')]
+        self.inner_src_ip_interval = lpm.LpmDict.IpInterval(ip_address(inner_src_ip_range[0]), ip_address(inner_src_ip_range[1]))
+        self.inner_dst_ip_interval = lpm.LpmDict.IpInterval(ip_address(inner_dst_ip_range[0]), ip_address(inner_dst_ip_range[1]))
+
+        outer_src_ip_range = [unicode(x) for x in self.test_params['outer_src_ip_range'].split(',')]
+        outer_dst_ip_range = [unicode(x) for x in self.test_params['outer_dst_ip_range'].split(',')]
+        self.outer_src_ip_interval = lpm.LpmDict.IpInterval(ip_address(outer_src_ip_range[0]), ip_address(outer_src_ip_range[1]))
+        self.outer_dst_ip_interval = lpm.LpmDict.IpInterval(ip_address(outer_dst_ip_range[0]), ip_address(outer_dst_ip_range[1]))
+
+        self.hash_keys = self.test_params.get('hash_keys', ['src-ip', 'dst-ip', 'src-port', 'dst-port'])
+        self.src_ports = self.test_params['src_ports']
+        self.symmetric_hashing = self.test_params.get('symmetric_hashing', False)
+        self.outer_dst_ip = self.outer_dst_ip_interval.get_first_ip()
+
+        self.next_hop = self.fib[self.outer_dst_ip]
+        self.exp_port_list = self.next_hop.get_next_hop_list()
+        assert (len(self.exp_port_list) > 1)
+        for exp_port in self.exp_port_list:
+            assert exp_port not in self.src_ports
+
+        self.balancing_range = self.test_params.get('balancing_range', self.DEFAULT_BALANCING_RANGE)
+        self.balancing_test_times = self.test_params.get('balancing_test_times', self.BALANCING_TEST_TIMES)
+
+
+    def check_hash(self, hash_key):
+        src_port = int(random.choice(self.src_ports))
+        logging.info("outer_dst_ip={}, src_port={}, exp_port_list={}".format(self.outer_dst_ip, src_port, self.exp_port_list))
+        outer_encap_formats = ['vxlan', 'nvgre']
+
+        for outer_encap_format in outer_encap_formats:
+            hit_count_map = {}
+            for _ in range(0, self.balancing_test_times*len(self.exp_port_list)):
+                logging.info('Checking {} hash key {}, src_port={}, exp_ports={}, dst_ip={}'\
+                    .format(outer_encap_format, hash_key, src_port, self.exp_port_list, self.outer_dst_ip))
+
+                ip_src = self.inner_src_ip_interval.get_random_ip() if hash_key == 'src-ip' else self.inner_src_ip_interval.get_first_ip()
+                ip_dst = self.inner_dst_ip_interval.get_random_ip() if hash_key == 'dst-ip' else self.inner_dst_ip_interval.get_first_ip()
+                sport = random.randint(0, 65535) if hash_key == 'src-port' else 1234
```

Comment:
@anish-n please avoid using magic numbers: `else 1234`

Reply:
Since this is a test, I personally feel ok with using hard coded values like the above. Besides this logic is almost identical to the currently implemented hash_test, I try to incorporate certain parts from the hash_test as-is so that we can leverage the existing thought process on hash testing. https://github.com/Azure/sonic-mgmt/blob/master/ansible/roles/test/files/ptftests/hash_test.py#L186